### PR TITLE
YAML Rule Editor Support

### LINKED ISSUE #154: [FEATURE] Create/edit rule in YAML format
**Is your feature request related to a problem?**
As a user, I should be able to create/edit a rule in YAML format or with a guided visual editor.

**What solution would you like?**
- In the Create rule flow, users have the option to define a rule using visual or YAML editor with a toggle.
- Users can toggle between editor and the rule will be translated. For example, if a user started creating the rule in YAML, then toggles over to visual editor, everything from the YAML editor will be translated into the visual editor. The user can continue editing the rule in visual editor.

**What alternatives have you considered?**
Open to suggestions!

**Mockups**
1. In the create rule flow, users can click on "YAML editor" to switch from Visual editor to YAML editor. (For toggle, use the [EuiButtonGroups](https://eui.elastic.co/v34.6.0/#/navigation/button#button-groups) component with "single" type)
image

2. Here's how it looks like when toggled to YAML editor. Users can click on "Visual editor" to return the define the rule using the UI. (The "Sigma specification" link should point to: https://github.com/SigmaHQ/sigma-specification)
image
[Updated screenshot] Added label and description text for the YAML editor: 
Label: Define rule in YAML
Description text: Use the YAML editor to define a sigma rule. See [Sigma specification](https://eui.elastic.co/v34.6.0/#/navigation/button#button-groups) for rule structure and schema.

### PR DESCRIPTION
**Pull request by djindjic**
### Description
Add YAML editor support

### Issues Resolved
#154 

### Check List
- [ ] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: public/pages/Rules/components/RuleEditor/RuleEditorFormState.model.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
+/*
```
Comment (amsiglan) [inline public/pages/Rules/components/RuleEditor/RuleEditorFormState.model.ts:1]: Let's just name this file as `RuleEditorFormState.ts`? Adding the model suffix would be a new pattern specific to this plugin and would be confusing.

**Review by amsiglan — APPROVED** [review]

File: cypress/integration/2_rules.spec.js [hunk 2]
```diff
@@ -93,6 +114,20 @@ describe('Rules', () => {
...
 
+      // Switch to YAML editor
+      cy.get(
+        '[data-test-subj="change-editor-type"] label:nth-child(2)',
+        TWENTY_SECONDS_TIMEOUT
+      ).click({
+        force: true,
+      });
+
+      YAML_RULE_LINES.forEach((line) =>
```
Comment (AWSHurneyt) [inline cypress/integration/2_rules.spec.js:125]: **Not blocking**, but could you make an `enhancement` issue to improve this portion of the test, and the check on [line 212](https://github.com/opensearch-project/security-analytics-dashboards-plugin/pull/201/files#diff-38d7c2fdd6835ad84ae7e1986f9e53e4269f0422c55ab8493c7ef842e89dde72R212)? I learned when implementing the `alerts.spec` test that the contents of the `EuiCodeEditor` is segmented into individual lines. [Here](https://github.com/opensearch-project/security-analytics-dashboards-plugin/blob/main/cypress/integration/3_alerts.spec.js#L278), I implemented a test which would iterate through each line to check whether it contained the expected string. Refactoring these tests to use a similar method would give us more confidence that the code editor contains the expected values in the expected format.

**Review by AWSHurneyt — APPROVED** [review]

File: cypress/integration/2_rules.spec.js (hunk 2 above)
Comment (djindjic) [reply cypress/integration/2_rules.spec.js:125]: Cool, thanks for sharing. Just create new issue on it, and will assign it to me in our task tracking
https://github.com/opensearch-project/security-analytics-dashboards-plugin/issues/208

Comment (AWSHurneyt) [reply cypress/integration/2_rules.spec.js:125]: Thank you!

**Pull request merged**